### PR TITLE
docs: Fix wrong condition for an app rejection

### PR DESCRIPTION
**Pull request by eugenesvk**
Swapped logic of rejection from work to failure and added an example

- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [ ] Have you successfully run `brew style` with your changes locally?
- [ ] Have you successfully run `brew typecheck` with your changes locally?
- [ ] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

File: docs/Acceptable-Casks.md [hunk 1]
```diff
@@ -118,7 +118,7 @@ Common reasons to reject a cask entirely:
 + The author has [specifically asked us not to include it](https://github.com/Homebrew/homebrew-cask/pull/5342).
 + App requires [SIP to be disabled](https://github.com/Homebrew/homebrew-cask/pull/41890) to be installed and/or used.
 + App installer is a `pkg` that requires [`allow_untrusted: true`](https://docs.brew.sh/Cask-Cookbook#pkg-allow_untrusted).
-+ App works on Homebrew supported macOS versions and platforms with GateKeeper enabled.
++ App fails with GateKeeper enabled on Homebrew supported macOS versions and platforms (e.g., unsigned apps fail on Macs with Apple silicon/ARM).
```
Comment (MikeMcQuaid) [inline docs/Acceptable-Casks.md:121]:
```suggestion
+ App fails with GateKeeper enabled on Homebrew supported macOS versions and platforms (e.g. unsigned apps fail on Macs with Apple Silicon/ARM).
```

**Review by MikeMcQuaid — APPROVED** [review]

File: docs/Acceptable-Casks.md (hunk 1 above)
Comment (eugenesvk) [reply docs/Acceptable-Casks.md:121]: Both of these are wrong: Apple doesn't capitalize silicon https://support.apple.com/en-us/HT211814 and AP Style book mandates a comma

Comment (MikeMcQuaid) [reply docs/Acceptable-Casks.md:121]: Fair on Silicon. We don't follow AP style here, though, but British English that doesn't use this form.

Comment (eugenesvk) [reply docs/Acceptable-Casks.md]: That also varies depending on the adopted style, eg, The Economist uses a nicer `eg,` with a comma, but without the periods

**Comment by MikeMcQuaid** [comment]
Thanks for the first PR, @eugenesvk! Great work!

**Pull request merged**